api/src/main/java/jakarta/enterprise/concurrent/ContextService.java

```diff
@@ -216,6 +217,45 @@ public interface ContextService {
    */
   public <R> Supplier<R> contextualSupplier(Supplier<R> supplier);
 
+  /**
+   * <p>Wraps a {@link java.util.concurrent.Flow.Subscriber} with context captured from the thread that invokes
+   * <code>contextualSubscriber</code>. Context is captured at the time <code>contextualSubscriber</code> is invoked.</p>
+   *
+   * <p>When one of the methods of the {@link java.util.concurrent.Flow.Subscriber} interface (<code>onSubscribe</code>,
+   * <code>onNext</code>, <code>onError</code>, or <code>onComplete</code>) is invoked on the proxy instance,
```

Comment:
Explicitly listing all of the methods that currently exist on Subscriber makes this match up with Java 21. We should also consider the possibility (which is probably unlikely, but it could happen) that a future version of Java adds another method to the interface.  As written, there would be no expectation to establish context for that method.  If we want to cover that possibility here, the wording could be changed as follows:

```suggestion
   * <p>Whenever one of the methods of the {@link java.util.concurrent.Flow.Subscriber} interface (such as <code>onSubscribe</code>,
   * <code>onNext</code>, <code>onError</code>, or <code>onComplete</code>) is invoked on the proxy instance,
```

Reply:
How would this work from a certification/TCK perspective? If we don't specify the methods, couldn't that result in a formerly compatible implementation becoming incompatible with the release of a new Java version, if the implementation isn't immediately updated to handle the new methods?

Reply:
That's actually the point here.  If an implementation wants to be compatible with a newer Java version than the TCK and spec were written for, it will need to handle the additional methods that the newer Java level introduces from the start.  By writing the requirement this way to include all interface methods rather than only the ones that happened to exist at the time, this makes it an unambiguous, non-breaking change moving forward to a future version of the Concurrency spec and TCK that specifies support for the newer Java level.

If we don't do this, then it would become a breaking change to start requiring context propagation on newSubscriberMethodX where it previously wasn't required when the spec moves to the newer Java level.

Reply:
Ok, I think that makes sense, so essentially without this we could encounter a situation like:

1. Java 23 releases and adds `Flow.Subscriber.methodX()`
2. User upgrades to Java 23 and uses Flow.Subscriber.methodX() in a contextualized Subscriber expecting that it doesn't have context
3. Jakarta EE 12 targets a Java version after 23, and so contextualizing `Flow.Subscriber.methodX()` would break the User in step 2. 

 I agree this should cover all current and future methods (even if they're unlikely).